crates/red_knot_python_semantic/src/types/display.rs

```diff
@@ -88,6 +88,24 @@ impl Display for DisplayRepresentation<'_> {
             },
             Type::KnownInstance(known_instance) => f.write_str(known_instance.repr(self.db)),
             Type::FunctionLiteral(function) => f.write_str(function.name(self.db)),
+            Type::Callable(CallableType::BoundMethod(bound_method)) => {
+                f.write_str("<bound method `")?;
+                f.write_str(bound_method.function(self.db).name(self.db))?;
+                f.write_str("` of `")?;
+                bound_method
+                    .self_instance(self.db)
+                    .display(self.db)
+                    .fmt(f)?;
+                f.write_str("`>")
```

Comment:
Nit: I would probably use `write!` here
```suggestion
                write!(
                    f, 
                    "<bound method `{method}` of `{instance}`>`",
                    method = bound_method.function(self.db).name(self.db),
                    instance = bound_method.self_instance(self.db).display(self.db)
                )?;
```

And the same for the other `Callable` handler